kpatch-build/create-diff-object.c

```diff
@@ -177,6 +177,10 @@ static struct rela *toc_rela(const struct rela *rela)
 	    rela->type != R_PPC64_TOC16_LO_DS)
 		return (struct rela *)rela;
 
+	/* Only constants in toc */
```

Comment:
Should we re-phrase the comment, saying older GCC has this issue?

Reply:
Are we 100% sure that having ".toc" section without a ".rela.toc" section never happens for new versions of GCC?

If so I can update the comment. Otherwise, I'd prefer keeping it as is.

Reply:
Let me try other GCC before conforming about it. We have always documented such powerpc quirks with comments, here I would suggest something like:
```
diff --git a/kpatch-build/create-diff-object.c b/kpatch-build/create-diff-object.c
index c9d2cc6..ded5fc1 100644
--- a/kpatch-build/create-diff-object.c
+++ b/kpatch-build/create-diff-object.c
@@ -177,6 +177,22 @@ static struct rela *toc_rela(const struct rela *rela)
            rela->type != R_PPC64_TOC16_LO_DS)
                return (struct rela *)rela;
 
+       /*
+        * it observed with certain objects, the .toc section's rela pair
+        * .rela.toc is not generated, when  the .toc section entries are
+        * only constants:
+        * objddump -dr net/ceph/pagelist.o:
+        * ...
+        * 0000000000000000 <.toc>:
+        *    0:   e0 ff ff ff     .long 0xffffffe0
+        *    4:   ff ff ff 0f     twui    r31,-1
+        *
+        * readelf -rW net/ceph/pagelist.o:
+        * ...
+        * [20] .data             PROGBITS        0000000000000000 000aa8 000000 00  WA  0   0  1
+        * [21] .toc              PROGBITS        0000000000000000 000aa8 000008 00  WA  0   0  8
+        * [22] ___kcrctab+ceph_pagelist_append PROGBITS        0000000000000000 000ab0 000008 00  WA  0   0  8
+        */
```